apps/pipeline-visualization/src/mermaid_utils.ts

```diff
@@ -0,0 +1,156 @@
+}
+
+export function createMermaidPipeline(pipeline: PipelineDefinition, mermaidOptions: MermaidOptions){
+  /* const myToString = (block: BlockDefinition, index = 0): string => {
+    const blockTypeName = block.type.ref?.name;
+    const blockString = `${index}["\`${block.name} \n (${blockTypeName})\`"]`;
+    const childString = collectChildren(block)
+      .map((child) => myToString(child, index + 1))
+      .join('-->');
+    return blockString + '-->' + childString;
+  };
+ */
+  let name: string = "subgraph " + pipeline.name;
+  let direction: string = "direction " + subgraphDirection
+  let listofPipes: Array<string[]> = [];
+  let listofBocks: Array<string> = [];
+  let composites: string = "";
+  const process_pipe = (pipe: string[], block:BlockDefinition) => {
+    if (block.type.ref?.$type == "CompositeBlocktypeDefinition" && mermaidOptions.composite){
+      let compositePipe: string[] = []
+      for (let subblock of block.type.ref?.blocks){
+        compositePipe.push(subblock.name)
+      }
+      let compositeName = "subgraph " + block.name;
+      composites += compositeName + "\n" + direction + "\n" + compositePipe.join("-->") + "\n" + "end \n";
+    }
+    pipe.push(block.name)
+    let propertyString = ""
+    if (mermaidOptions.properties){
+      for (let property of block.body.properties) {
+        if (properties.includes(property.name)){
+          let pv: any = property.value // Is there a better way of doing this?
+          propertyString += `${property.name}: ${pv.value}\n`
+        }
+      }
+    }
+    //let propertyString = block.body.properties.map((property) => `${property.name}: ${property.value.value}`).join("\n")
+    //Markdown styling
+    //listofBocks.push(`${block.name}["\` ${block.name}\n(${block.type.ref?.name})\n${propertyString}\`" ]`)
+    //HTML styling
+    listofBocks.push(`${block.name}[${block.name}<br><i>${block.type.ref?.name}</i><br>${propertyString}]`)
+    let children = collectChildren(block);
+    if (children.length == 1) {
+      process_pipe(pipe, children[0]!);
+    } else if (children.length > 1){
+      listofPipes.push(pipe);
+      children.forEach((child) => {process_pipe([block.name], child)})
+    } else {
+      listofPipes.push(pipe)
+    }
+  }
+  let startingBlocks = collectStartingBlocks(pipeline);
+  startingBlocks.forEach((startingBlock) => process_pipe([], startingBlock))
+  let pipelineSet = listofPipes
+  .map((pipeline) => pipeline.join("-->"))
+  .join("\n")
+  let blockSet = listofBocks.join("\n")
+   
+  return name + "\n" + direction + "\n" + pipelineSet + "\n" + "end \n" + composites + "\n" + blockSet + "\n";
+  }
+
+export function createMermaidStyling(pipeline: PipelineDefinition) {
+  let classAssign = []
+  for (const block of pipeline.blocks){
+    if (block.name.includes("Extractor")){
+      classAssign.push(`class ${block.name} source;`)
+    }
+    if (block.name.includes("Loader")){
+      classAssign.push(`class ${block.name} sink;`)
+    }
```

Comment:
The block names are arbitrary, i.e. it is possible to name an extractor `MyCoolLoader`. You should use our utility functions (in this case the attribute `inputType` of a  `BlockTypeWrapper` could help)